server/chat-plugins/chatlog.ts

```diff
@@ -666,17 +687,135 @@ export const LogSearcher = new class {
+		const args = [user ? '--count' : ''];
+		const response = await this.ripgrepSearchMonth({
+			search: regexString, raw: true, date: month, room, args,
+		});
+		const rawResults = response.results.shift();
+		if (!rawResults) return LogViewer.error(`No results found.`);
+		const results: {[k: string]: {[userid: string]: number}} = {};
+		for (const fullLine of rawResults.split('\n')) {
+			const [data, line] = fullLine.split('.txt:');
+			const date = data.split('/').pop()!;
+			if (!results[date]) results[date] = {};
+			if (!toID(date)) continue;
+			if (user) {
+				if (!results[date][user]) results[date][user] = 0;
+				const parsed = parseInt(line);
+				results[date][user] += isNaN(parsed) ? 0 : parsed;
+			} else {
+				const parts = line?.split('|').map(toID);
+				if (!parts || parts[1] !== 'c') continue;
+				const id = parts[2];
+				if (!id) continue;
+				if (!results[date][id]) results[date][id] = 0;
+				results[date][id]++;
+			}
+		}
+		return this.renderLinecountResults(results, room, month, user);
+	}
+	async fsSearchLinecounts(roomid: RoomID, month: string, user?: ID) {
+		const directory = FS(`logs/chat/${roomid}/${month}`);
+		if (!directory.existsSync()) {
+			throw new Chat.ErrorMessage(`Logs for month '${month}' do not exist on room ${roomid}.`);
+		}
+		const files = await directory.readdir();
+		const results: {[date: string]: {[userid: string]: number}} = {};
+		for (const file of files) {
+			const day = file.slice(0, -4);
+			const stream = FS(`logs/chat/${roomid}/${month}/${file}`).createReadStream();
+			for await (const line of stream.byLine()) {
+				const parts = line.split('|').map(toID);
+				const id = parts[2];
+				if (!id) continue;
+				if (parts[1] === 'c') {
+					if (user && id !== user) continue;
+					if (!results[day]) results[day] = {};
+					if (!results[day][id]) results[day][id] = 0;
+					results[day][id]++;
+				}
+			}
+		}
+		return this.renderLinecountResults(results, roomid, month, user);
+	}
+	renderLinecountResults(
+		results: {[date: string]: {[userid: string]: number}}, roomid: RoomID,
+		month: string, user?: ID
+	) {
+		let buf = Utils.html`<div class="pad"><h2>Linecounts on `;
+		buf += `${roomid}${user ? ` for the user ${user}` : ` (top ${MAX_TOPUSERS})`}</h2>`;
+		buf += `<strong>Month: ${month}:</strong><br />`;
+		const nextMonth = LogReader.nextMonth(month);
+		const prevMonth = LogReader.prevMonth(month);
+		if (FS(`logs/chat/${roomid}/${prevMonth}`).existsSync()) {
+			buf += `<small><a roomid="view-roomstats-${roomid}--${prevMonth}${user ? `--${user}` : ''}">Previous month</a></small>`;
+		}
+		if (FS(`logs/chat/${roomid}/${nextMonth}`).existsSync()) {
+			buf += ` <small><a roomid="view-roomstats-${roomid}--${nextMonth}${user ? `--${user}` : ''}">Next month</a></small>`;
+		}
+		buf += `<hr /><ol>`;
+		if (user) {
+			const sortedDays = Object.keys(results).sort((a, b) => {
+				return new Date(b).getTime() - new Date(a).getTime();
+			});
```

Comment:
```suggestion
			const sortedDays = Object.keys(results).sort((a, b) => (
				new Date(b).getTime() - new Date(a).getTime()
			));
```

I feel like I've suggested this a lot lately.